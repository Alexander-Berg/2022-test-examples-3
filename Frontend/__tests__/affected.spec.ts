import { Affected } from '..';
import { lerna, FakeVcsEmpty, FakeVcsOnlyAffected, FakeVcsAffected, FakeVcsService, root, packages } from '../../__mocks__';

describe('affected: Возвращает корректный список измененных пакетов', () => {
    beforeEach(() => {
        jest.resetModules();
    });

    test('Возвращает пустые directs, dependent, если ничего не изменилось', async() => {
        const service = new FakeVcsEmpty();
        const { directs, sortedAffectedPackages } = await new Affected({
            vcs: { service },
            lerna,
            root,
        }).execute();
        expect(directs.size).toEqual(0);
        expect(sortedAffectedPackages.length).toEqual(0);
    });

    describe('npm пакет', () => {
        test('Пакет нигде не используется – Возвращаем только он сам', async() => {
            const service = new FakeVcsOnlyAffected();
            jest.mock('/root/packages/emptydeps/package.json', () => { return { name: 'empty' } }, { virtual: true });
            const { directs, sortedAffectedPackages } = await new Affected({
                vcs: { service },
                lerna,
                root,
            }).execute();
            expect(directs.size).toEqual(1);
            expect(directs).toMatchSnapshot('empty deps');
            expect(sortedAffectedPackages.length).toEqual(1);
        });
        test('Пакет используется в других местах – Возвращаем пакет и того, кто он него зависит', async() => {
            const service = new FakeVcsAffected();
            jest.mock('/root/packages/ui/package.json', () => { return packages['@yandex/ui'] }, { virtual: true });
            jest.mock('/root/services/service/package.json', () => { return packages.service }, { virtual: true });
            jest.mock('/root/packages/emptydeps/package.json', () => { return packages.empty }, { virtual: true });
            jest.mock('/root/packages/dependant/package.json', () => { return packages.empty }, { virtual: true });
            const { directs, sortedAffectedPackages } = await new Affected({
                vcs: { service },
                lerna,
                root,
            }).execute();
            expect(directs.size).toEqual(1);
            expect(directs).toMatchSnapshot('ui');
            expect(sortedAffectedPackages.length).toEqual(Object.keys(packages).length);
            expect(sortedAffectedPackages).toMatchSnapshot('ui deps');
        });
    });

    describe('Приватный пакет', () => {
        test('Возвращает только один пакет', async() => {
            const service = new FakeVcsService();
            jest.mock('/root/services/service/package.json', () => { return packages.service }, { virtual: true });
            const { directs, sortedAffectedPackages } = await new Affected({
                vcs: { service },
                lerna,
                root,
            }).execute();
            expect(directs.size).toEqual(1);
            expect(directs).toMatchSnapshot('service');
            expect(sortedAffectedPackages.length).toEqual(1);
        });

        test('Возвращает только один пакет --noPrivate ', async() => {
            const service = new FakeVcsService();
            jest.mock('/root/services/service/package.json', () => { return packages.service }, { virtual: true });
            const { directs, sortedAffectedPackages } = await new Affected({
                vcs: { service },
                lerna,
                root,
                noPrivate: true,
            }).execute();
            expect(directs.size).toEqual(1);
            expect(directs).toMatchSnapshot('service noPrivate');
            expect(sortedAffectedPackages.length).toEqual(1);
        });
    });
});
