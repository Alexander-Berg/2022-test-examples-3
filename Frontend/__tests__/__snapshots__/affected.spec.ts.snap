// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`affected: Возвращает корректный список измененных пакетов npm пакет Пакет используется в других местах – Возвращаем пакет и того, кто он него зависит: ui 1`] = `
Map {
  "@yandex/ui" => Object {
    "autobumpversion": undefined,
    "directCommits": Array [
      Object {
        "files": Array [
          "/packages/ui/package.json",
        ],
        "ignore": false,
        "message": "fix(ui): update prod deps",
        "sha": "shabranch",
      },
    ],
    "location": "/root/packages/ui",
    "mergeCommits": Array [
      Object {
        "directCommits": Array [
          Object {
            "files": Array [
              "/packages/ui/package.json",
            ],
            "ignore": false,
            "message": "build(ui): update pack",
            "sha": "pr1hash",
          },
        ],
        "merge": "Merge pull request #1 from fork/branch",
        "prTitle": "ISL-123: Обновили @bem-react/pack",
        "sha": "mergesha",
      },
    ],
    "name": "@yandex/ui",
    "private": false,
    "tag": "npm",
    "version": "1.0.0",
  },
}
`;

exports[`affected: Возвращает корректный список измененных пакетов npm пакет Пакет используется в других местах – Возвращаем пакет и того, кто он него зависит: ui deps 1`] = `
Array [
  Object {
    "autobumpversion": undefined,
    "directCommits": Array [
      Object {
        "files": Array [
          "/packages/ui/package.json",
        ],
        "ignore": false,
        "message": "fix(ui): update prod deps",
        "sha": "shabranch",
      },
    ],
    "location": "/root/packages/ui",
    "mergeCommits": Array [
      Object {
        "directCommits": Array [
          Object {
            "files": Array [
              "/packages/ui/package.json",
            ],
            "ignore": false,
            "message": "build(ui): update pack",
            "sha": "pr1hash",
          },
        ],
        "merge": "Merge pull request #1 from fork/branch",
        "prTitle": "ISL-123: Обновили @bem-react/pack",
        "sha": "mergesha",
      },
    ],
    "name": "@yandex/ui",
    "private": false,
    "tag": "npm",
    "version": "1.0.0",
  },
  Object {
    "autobumpversion": undefined,
    "dependencies": Object {
      "@yandex/ui": "1.0.0",
    },
    "depsAffected": true,
    "location": "/root/services/service",
    "name": "service",
    "private": true,
    "tag": "permanent",
    "version": "1.0.0",
  },
  Object {
    "autobumpversion": false,
    "dependencies": Object {
      "@yandex/ui": "1.0.0",
    },
    "depsAffected": true,
    "location": "/root/packages/emptydeps",
    "name": "empty",
    "private": false,
    "tag": "permanent",
    "version": "0.0.0",
  },
  Object {
    "autobumpversion": false,
    "dependencies": Object {
      "@yandex/ui": "1.0.0",
    },
    "depsAffected": true,
    "location": "/root/packages/dependant",
    "name": "empty",
    "private": false,
    "tag": "permanent",
    "version": "0.0.0",
  },
]
`;

exports[`affected: Возвращает корректный список измененных пакетов npm пакет Пакет нигде не используется – Возвращаем только он сам: empty deps 1`] = `
Map {
  "empty" => Object {
    "autobumpversion": undefined,
    "directCommits": Array [
      Object {
        "files": Array [
          "/packages/emptydeps/package.json",
        ],
        "ignore": false,
        "message": "fix(emptydeps): update prod deps",
        "sha": "shabranch",
      },
    ],
    "location": "/root/packages/emptydeps",
    "mergeCommits": Array [
      Object {
        "directCommits": Array [
          Object {
            "files": Array [
              "/packages/emptydeps/package.json",
            ],
            "ignore": false,
            "message": "chore(emptydeps): add owners",
            "sha": "pr1hash",
          },
        ],
        "merge": "Merge pull request #1 from fork/branch",
        "prTitle": "TRIVIAL – test",
        "sha": "mergesha",
      },
    ],
    "name": "empty",
    "private": false,
    "tag": "npm",
    "version": undefined,
  },
}
`;

exports[`affected: Возвращает корректный список измененных пакетов Приватный пакет Возвращает только один пакет --noPrivate : service noPrivate 1`] = `
Map {
  "service" => Object {
    "autobumpversion": undefined,
    "directCommits": Array [],
    "location": "/root/services/service",
    "mergeCommits": Array [],
    "name": "service",
    "private": true,
    "tag": "permanent",
    "version": "1.0.0",
  },
}
`;

exports[`affected: Возвращает корректный список измененных пакетов Приватный пакет Возвращает только один пакет: service 1`] = `
Map {
  "service" => Object {
    "autobumpversion": undefined,
    "directCommits": Array [
      Object {
        "files": Array [
          "/services/service/package.json",
        ],
        "ignore": false,
        "message": "fix(emptydeps): update prod deps",
        "sha": "shabranch",
      },
    ],
    "location": "/root/services/service",
    "mergeCommits": Array [
      Object {
        "directCommits": Array [
          Object {
            "files": Array [
              "/services/service/package.json",
            ],
            "ignore": false,
            "message": "chore(service): add owners",
            "sha": "pr1hash",
          },
        ],
        "merge": "Merge pull request #1 from fork/branch",
        "prTitle": "TRIVIAL – test",
        "sha": "mergesha",
      },
    ],
    "name": "service",
    "private": true,
    "tag": "permanent",
    "version": "1.0.0",
  },
}
`;
